### PR TITLE
Update for 3p notices and json file as of `o3de/47af914`

### PR DESCRIPTION
**Pull request by amzn-changml**
This is a large diff due to a different ordering of the license files and the lag between release in July till now. Includes engine and 3p folder scans.

Signed-off-by: Mike Chang <changml@amazon.com>

### CODE REVIEW

**Review by brianherrera — APPROVED** [review]
Just a general workflow question. Should this update be pushed to (or generated from) the latest stabilization branch which will then be merged to main once the release is ready? I know it will be pushed to main eventually.

**Comment by amzn-changml** [comment]
> Just a general workflow question. Should this update be pushed to (or generated from) the latest stabilization branch which will then be merged to main once the release is ready? I know it will be pushed to main eventually.

I was actually looking to target Dev, as 3p updates tend to get pushed there first. Happy to take suggestions otherwise, I'm looking to create a RFC to handle the proposal. 

I hope to have some sort of cadence in which this is PR'ed; the proposed automation will scan https://github.com/o3de/o3de/tree/development/cmake/3rdParty/Platform for changes, so we could support a PR on every change. This is mostly to support a CD of the installer.

**Comment by lawsonamzn** [comment]
Looks good to me!  Does this still need to be moved forward?

**Review by lawsonamzn — APPROVED** [review]

File: SPDX-Licenses.json [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+        "License": "Zlib",
+        "LicenseFile": "zlib/LICENSE"
+    },
+    {
+        "PackageName": "zstd-1.35-multiplatform",
+        "URL": "http://www.zstd.net/",
+        "License": "BSD",
+        "LicenseFile": "zstd/LICENSE"
+    }
+]
```
Comment (AMZN-alexpete) [inline SPDX-Licenses.json:278]: minor, but would it hurt to add a newline at the end for linux?

Comment (amzn-changml) [reply SPDX-Licenses.json:278]: Sure, I can add it here and fix it in the `license_scan.py` script

**Comment by amzn-changml** [comment]
> 
> 
> Looks good to me! Does this still need to be moved forward?

Yes, this is also validating the output of the `license_scan.py` script. I'll commit this in once we're all good.

**Pull request merged**